calico_rkt/calico_rkt.py

```diff
@@ -1,138 +1,150 @@
+import logging
 import json
 import os
 import sys
 from subprocess import check_output, CalledProcessError, check_call
-
 from netaddr import IPAddress, IPNetwork, AddrFormatError
+
+from logutils import configure_logger
 from pycalico import netns
 from pycalico.ipam import IPAMClient, SequentialAssignment
 from pycalico.netns import Namespace
-from pycalico.datastore_datatypes import Rules
+from pycalico.datastore_datatypes import Rules, IPPool
 from pycalico.datastore import IF_PREFIX, DatastoreClient
 from pycalico.datastore_errors import PoolNotFound
 
-print_stderr = functools.partial(print, file=sys.stderr)
+_log = logging.getLogger(__name__)
 
-
-input_ = ''.join(sys.stdin.readlines()).replace('\n', '')
-input_json = json.loads(input_)
-
-# Append to existing env, to avoid losing PATH etc.
-# TODO-PAT: This shouldn't be hardcoded
-env = os.environ.copy()
 ETCD_AUTHORITY_ENV = 'ETCD_AUTHORITY'
-env[ETCD_AUTHORITY_ENV] = 'localhost:2379'
-
-# PROFILE_LABEL = 'CALICO_PROFILE'
-# ETCD_PROFILE_PATH = '/calico/'
 
 ORCHESTRATOR_ID = "rkt"
 HOSTNAME = socket.gethostname()
-NETNS_ROOT= '/var/lib/rkt/pods/run'
+NETNS_ROOT = '/var/lib/rkt/pods/run'
 
-def main():
+def main(env, conf_in):
     mode = env['CNI_COMMAND']
 
-    if mode == 'init':
-        print_stderr('No initialization work to perform')
-    elif mode == 'ADD':
-        print_stderr('Executing Calico pod-creation plugin')
-        create(
-            container_id=env['CNI_CONTAINERID']
-            )
+    if mode == 'ADD':
+        create(env=env, conf_in=conf_in)
     elif mode == 'DEL':
-        print_stderr('Executing Calico pod-deletion plugin')
-        delete(
-            container_id=env['CNI_CONTAINERID']
-            )
+        delete(env=env, conf_in=conf_in)
 
-def create(container_id):
+def create(env, conf_in):
     """"Handle rkt pod-create event."""
-    print_stderr('Configuring pod %s' % container_id, file=sys.stderr)
+    container_id = env['CNI_CONTAINERID']
+
+    _log.info('Configuring pod %s' % container_id)
     netns_path='%s/%s/%s' % (NETNS_ROOT, container_id, env['CNI_NETNS'])
     _datastore_client = IPAMClient()
 
     try:
         endpoint, ip = _create_calico_endpoint(container_id=container_id,
-                                            netns_path=netns_path,
-                                            client=_datastore_client)
+                                               netns_path=netns_path,
+                                               client=_datastore_client,
+                                               conf_in=conf_in,
+                                               interface = env['CNI_IFNAME'])
 
-        _create_profile(endpoint=endpoint, 
-                        profile_name=input_json['name'], 
+        _set_profile_on_endpoint(endpoint=endpoint,
+                        profile_name=conf_in['name'],
                         ip=ip,
                         client=_datastore_client)
     except CalledProcessError as e:
```

Comment:
You should split these two function calls each into their own try/except blocks. It ensures that your only catching the errors you expect each to call. 

Even if in this situation its possible that both functions throw the error, we'll still want them split so later on we can add teardown if the second fails but not the first. e.g. the except block for _set_profile_on_endpoint will eventually need to delete the endpoint (but thats something we'll implement later)